docs/data-migration.md

```diff
@@ -0,0 +1,268 @@
+the chunk size during the migration as well.
+If you want to start the migration from the beginning (can only be done if the
+migration did still not succeed) use the flag `--force-new-migration` which can
+be used in combination with a new `chunksize` limit.
+
+
+## After the migration was successful
+
+Make sure the whole migration process succeeds before starting the new node with
+the new underlying database. As mentioned above there are several database files
+at play here so all of them have to succeed to garantee a successful migration.
+
+If the migration succeeded successfully and you see the following log entry of
+the migration tool you are good to go to start-up your lnd node with the new
+database.
+
+```shell
+[INF]: LNDINIT !!!Migration of all mandatory db parts completed successfully!!!
+```
+
+The mandatory dbs are:
+* `channel.db`
+* `macaroons.db`
+* `sphinxreplay.db`
+* `wallet.db`
+
+The optional dbs are:
+
+* `wtclient.db`
+* `watchtower.db`
+* `neutrino.db`
+
+### LND config setting for `sqlite` backend
+
+```shell
+[db]
+db.backend=sqlite
+```
+
+There are several other sqlite setttings you can tweak to make it fit your 
+needs, take a look at the [lnd-sample-config](https://github.com/lightningnetwork/lnd/blob/b6d8ecc7479f7517368814c398b0fbe0e8c52fed/sample-lnd.conf).
+
+### LND config setting for `postgres` backend
+
+```shell
+[db]
+db.backend=postgres
+
+[postgres]
+
+db.postgres.dsn=postgres://xx:yy@localhost:5432/zz
+```
+
+Use the same connection string you used for the migration. Also take a look at 
+the postgres knobs in the [lnd-sample-config](https://github.com/lightningnetwork/lnd/blob/b6d8ecc7479f7517368814c398b0fbe0e8c52fed/sample-lnd.conf).
+
+
+This is the output of the migration cmd help settings:
+
+```shell
+lndinit migrate-db -h
+2025-04-01 00:29:24.618 [INF]: LNDINIT Version 0.1.24-beta commit=docker/v0.1.22-beta-lnd-v0.18.3-beta.rc1-10-g658c5c8-dirty, debuglevel=
+Usage:
+  lndinit [OPTIONS] migrate-db [migrate-db-OPTIONS]
+
+Migrate the full database state of lnd from a source (for example the
+set of bolt database files such as channel.db and wallet.db) database
+to a SQL destination database.
+
+IMPORTANT: Please read the data migration guide located	in the file
+docs/data-migration.md of the main lnd repository before using this
+command!
+
+NOTE: The migration can take a long time depending on the amount of data
+that needs to be written! The migration happens in chunks therefore it
+can be resumed in case of an interruption. The migration also includes
+a verification to assure that the migration is consistent.
+As long as NEITHER the source nor destination database has been started/
+run with lnd, the migration can be repeated/resumed in case of an error
+since the data will just be overwritten again in the destination.
+
+Once a database was successfully and completely migrated from the source
+to the destination, the source will be marked with a 'tombstone' tag
```

Comment:
I think we should add a check in LND that if this tombstone is set then LND won't start up with the old backend settings.

Reply:
there is already a check in place in LND: 

https://github.com/lightningnetwork/lnd/blob/867d27d68a2a4e83a7d101090497f126794d1269/channeldb/db.go#L478-L480